jobs/involuntary-dissolutions/involuntary_dissolutions.py

```diff
@@ -238,6 +238,26 @@ def stage_2_process(app: Flask):
     stage_2_delay = timedelta(days=app.config.get('STAGE_2_DELAY'))
 
     for batch_processing in batch_processings:
+        # Check if email or letter furnishing entry has been completed. If not, do not transition to stage 2.
+        furnishings = Furnishing.find_by(
+            batch_id=batch_processing.batch_id,
+            business_id=batch_processing.business_id
+        )
+        furnishing_entry_completed = any(
+            furnishing.furnishing_type in (Furnishing.FurnishingType.EMAIL, Furnishing.FurnishingType.MAIL)
+            and furnishing.status == Furnishing.FurnishingStatus.PROCESSED
```

Comment:
So here, if the furnishing entry created after the furnishings job run is NOT processed, the `batch_processing` entry will not move forward to stage 2. It'll be of status `ERROR`.